projects/graph/src/graph.py

```diff
@@ -85,3 +85,10 @@ def dft(self, starting_vertex_id):
               # Then, put all of it's children into the stack
               for neighbor in self.vertices[v]:
                   s.push(neighbor)
+    def dft_recursive(self, starting_vertex_id, visited = {}):
```

Comment:
Good idea to define visited as a dictionary type, but since we only care about whether a vertex is in visited or not, you don't really need to the key-value relationship.  You can simply define visited as a set type, and then see if visited[starting_vertex_id] is true or false.